ecommerce-product-page-main/ecommerce/src/components/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,64 @@
+import {useState} from 'react';
+import styled from 'styled-components';
+
+import Card from '../Card/Card';
+import ProductThumbnail from './ProductThumbnail';
+import { S as Styled } from './FullCarousel';
+
+interface CarouselStyledProps {
+    isLeft?: boolean,
+    isRight?: boolean
+}
+
+interface CarouselComponentProps {
+    setShowCarousel: Function
+}
+
+export const S = {
+    CarouselContainer: styled.div`
+        margin: 0 auto;
+        position: relative;
+
+        @media all and (max-width: 770px) {
+            width: 100%;
+        }
+    `,
+    MobileCarouselControlBtn: styled(Styled.CarouselControlBtn)`
+        position: absolute;
+        top: 50%;
+        border: 1px solid var(--grayish-blue);
+        transform: translateY(-150%);
+        display: none;
+        left: ${({isLeft}: CarouselStyledProps) => isLeft ? "0": "initial"};
+        right: ${({isRight}: CarouselStyledProps) => isRight ? "0": "initial"};
+
+        @media all and (max-width: 770px) {
+            display: flex;
+        }
+    `,
+    MobileCarouselControlIcon: styled.img``,
```

Comment:
Is this necessary?

Reply:
I use it twice on the component, so I thought it would only make sense to put it there to make it less repetitive.